### PR TITLE
docs(api): fix return type of `nvim[_buf]_get_keymap()`

### LINKED ISSUE #31691: Wrong lua type annotation for the return value of `nvim_get_keymap` and `nvim_buf_get_keymap`
### Problem

`nvim_get_keymap` and `nvim_buf_get_keymap` return a list of mapping definitions which has the following fields:

- abbr
- buffer
- desc
- expr
- lhs
- lhsraw
- lhsrawalt
- lnum
- mode
- mode_bits
- noremap
- nowait
- rhs
- script
- scriptversion
- sid
- silent

However in `runtime/lua/vim/_meta/api.lua` they are annotated to return `vim.api.keyset.keymap[]` with extra fields `callback`, `replace_keycodes`, `unique` and is missing fields `abbr`, `buffer`, `lhs`, `lhsraw`, `lhsrawalt`, `lnum`, `mode`, `mode_bits`, `rhs`, `scriptversion`, and `sid`

### Steps to reproduce

N/A

### Expected behavior

`nvim_get_keymap` and `nvim_buf_get_keymap` should have correct type annotation in `_meta/api.lua`.

### Nvim version (nvim -v)

0.10.3

### Vim (not Nvim) behaves the same?

N/A

### Operating system/version

Linux

### Terminal name/version

Alacritty

### $TERM environment variable

tmux-256color

### Installation

AUR

### PR DESCRIPTION
**Pull request by Bekaboo**
Shoud fix #31691

### CODE REVIEW

File: runtime/lua/vim/_meta/api_keysets_extra.lua [hunk 1]
```diff
@@ -173,6 +173,26 @@ error('Cannot require a meta file')
 --- @field force? true
 --- @field cterm? vim.api.keyset.hl_info.cterm
 
+--- @class vim.api.keyset.get_keymap
+--- @field abbr? 0|1
```
Comment (glepnir) [inline runtime/lua/vim/_meta/api_keysets_extra.lua:177]: does there shouldn't boolean? and in doc says is return maparg() like dict maybe we can also link this object for maparg?

Comment (zeertzjq) [reply runtime/lua/vim/_meta/api_keysets_extra.lua:177]: No, they are integers

Comment (Bekaboo) [reply runtime/lua/vim/_meta/api_keysets_extra.lua:177]: I checked the result of `:lua =vim.api.nvim_get_keymap('n')` and found out that fields that should be boolean are represented by 0/1

Comment (Bekaboo) [reply runtime/lua/vim/_meta/api_keysets_extra.lua:177]: > maybe we can also link this object for maparg?

`maparg()` can returns the rhs of a mapping if `{dict}` if falsy or not specified, e.g. `maparg('<C-l>')`. Currently it is annotated to return a string.

We can do this for `maplist()` though which gives the same result as `nvim_get_keymap`

**Comment by Bekaboo** [comment]
@glepnir Is it a good idea to set `vim.api.keyset.get_keymap[]` as the return value of vimscript `maparg()` or `maplist()`? Different from `nvim_get_keymap()`, they are not API functions. The type annotations are generally more vague for the vimscript function compared to the detailed annotations for the `nvim_` API functions, and according to `vimfn.lua` there's no other vimcript function has type annotations of `vim.api.keyset.*`

**Comment by glepnir** [comment]
Not sure 😮‍💨. Just saw the documentation described it as maparg() like .  Maybe that's good enough for now.

**Comment by justinmk** [comment]
Obvious question is why not enhance `vim.api.keyset.keymap` instead of defining a new type? But I see that the new type has integer values whereas `vim.api.keyset.keymap` has booleans. 

Then the next question is WHY `nvim_buf_get_keymap` doesn't coerce stuff to booleans?

**Review by justinmk — APPROVED** [review]
Meh.

**Comment by Bekaboo** [comment]
> Then the next question is WHY `nvim_buf_get_keymap` doesn't coerce stuff to booleans?

@justinmk Won't that break the API?

**Comment by justinmk** [comment]
> I'd like that too, but won't that break the API?

Maybe. That's a good reminder of why "roundtripping" is important: it helps discover this kind of thing from the start. Roundtrips are also discussed in https://github.com/neovim/neovim/pull/29464 @max397574 though I need to review if it's relevant here (the "coerce to boolean" topic).

**Comment by Bekaboo** [comment]
Adding `vim.keymap.get` indeed ease the urgent to change the return type of `nvim_get_keymap()` and avoid breaking the API.

**Pull request merged**